### PR TITLE
feature: ActionBlock with InlineLink

### LINKED ISSUE #218: Add action (button) group
> [!IMPORTANT] 
> Needs refinement

## Concept

Add action (button) groups so content editors can link to pages (and other models) and give those links a style (select from list "default", "primary", "default"). Always use them within a group, so we have more control over the layout in the UI, can add logic to only style a single link per group as a primary button, etc.

```
┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄
┆ ╔═════════════╗  ┌─────────────┐            ┆
┆ ║   Primary   ║  │  Secondary  │   Default  ┆  = action group
┆ ╚═════════════╝  └─────────────┘   ───────  ┆
┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄
```

## Questions

- Should we allow a max number of items per group? (or leave that up to developers using Head Start for their project?)
- ...

## Prior art

Actions (without group) in No Dodos website (based on Head Start) ([code](https://github.com/No-Dodos/nododos-website/blob/main/src/blocks/InternalLink/InternalLinkBlock.astro#L15)):

[UI](https://www.nododos.nl/en/) | CMS
--- | ---
image | image

### PR DESCRIPTION
**Pull request by jbmoelker**
# Changes

- Adds an Action Block model & component that can be used to add one or more prestyled actions (only Inline Link for now) to pages or other content blocks.
- Fixes the `npm create:...` (Plop) files so that test files are correctly added when creating a new Block or Component. (Not technically part of this PR, but I needed it to create the new blocks anyway).

```
┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄
┆ ╔═════════════╗  ┌─────────────┐            ┆
┆ ║   Primary   ║  │  Secondary  │   Default  ┆  = action group
┆ ╚═════════════╝  └─────────────┘   ───────  ┆
┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄┄
```

# Associated issue

Resolves #218

# How to test

1. Open preview link
2. Verify the home page has an ActionBlock above the fold and that it works
3. Navigate to the ActionBlock Demo page (`/en/demos/action-block-demo/`)
4. Verify that all the variations work

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have made sure that my PR is easy to review (not too big, includes comments)
- [x] I have made updated relevant documentation files (in project README, docs/, etc)
- ~I have added a decision log entry if the change affects the architecture or changes a significant technology~
- [x] I have notified a reviewer

### CODE REVIEW

File: src/blocks/ActionBlock/action.css [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/* basic styling, can be replaced */
```
Comment (jbmoelker) [inline src/blocks/ActionBlock/action.css:1]: Should this file be here, in the common container block for all actions, or should we consider this a core style and move it to `src/assets/action.css`?

File: config/datocms/migrations/1733671970_actionBlock.ts [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    },
+    {
+      skip_menu_item_creation: true,
+      schema_menu_item_id: 'aVOBPryiQ8O5EiHEHDEang',
+    }
+  );
+
+  console.log('Creating new fields/fieldsets');
+
+  console.log(
```
Comment (jbmoelker) [inline config/datocms/migrations/1733671970_actionBlock.ts:42]: Should an ActionBlock be able to contain any number of actions (1+) with any style, like defined here? Or do we prefer a model where an ActionBlock has single modular fields for `primaryAction`, `secondaryAction` and `defaultAction` (all optional)?

File: src/blocks/ActionBlock/ActionBlock.astro [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  warningText && (
+    <script is:inline define:vars={{ warningText }}>
+      console.warn(warningText);
+    </script>
+  )
+}
+
+<style>
+  /* basic styling, can be replaced */
+  .action-block {
```
Comment (jbmoelker) [inline src/blocks/ActionBlock/ActionBlock.astro:51]: What do we want as basic styling for this component?

**Review by jurgenbelien — APPROVED** [review]
👍

**Pull request merged**